ProteoformSuiteGUI/ExperimentTheoreticalComparison.cs

```diff
@@ -30,28 +31,61 @@ public ExperimentTheoreticalComparison()
         }
 
         public void ExperimentTheoreticalComparison_Load(object sender, EventArgs e)
-        { }
+        {
+        }
 
         public void compare_et()
         {
             if (Lollipop.et_relations.Count == 0 && Lollipop.proteoform_community.has_e_and_t_proteoforms)
             {
+                ClearListsAndTables();
                 run_the_gamut();
             }
             else if (Lollipop.et_relations.Count == 0 && Lollipop.proteoform_community.has_e_proteoforms) MessageBox.Show("Go back and create a theoretical database.");
             else if (Lollipop.et_relations.Count == 0) MessageBox.Show("Go back and aggregate experimental proteoforms.");
         }
 
+
         public void run_the_gamut()
         {
+            if (Lollipop.notch_search_et)
+            {
+                bool notch_masses = get_notch_masses();
+                if (!notch_masses) return;
+            }
+            ClearListsAndTables();
             this.Cursor = Cursors.WaitCursor;
             Lollipop.make_et_relationships(Lollipop.proteoform_community);
             ((ProteoformSweet)MdiParent).proteoformFamilies.ClearListsAndTables();
             this.FillTablesAndCharts();
+            if (Lollipop.ed_relations.Count > 0) cb_view_ed.Enabled = true;
             this.Cursor = Cursors.Default;
             compared_et = true;
         }
 
+        public bool get_notch_masses()
```

Comment:
(Oops, that reply was for another comment.) Sounds good. An easier option is to put these shared methods in a utility class in the ProteoformSuiteGUI project.